libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -30,6 +30,8 @@ declareCounter(libp2p_pubsub_validation_failure, "pubsub failed validated messag
 declarePublicCounter(libp2p_pubsub_messages_published, "published messages", labels = ["topic"])
 
 type
+  PeerTable* = Table[string, HashSet[PubSubPeer]]
```

Comment:
I would actually use the `PeerID` object here instead of `PubSubPeer`, peerid is guaranteed to be cryptographically unique.

Reply:
well, it's a bit the point of ref types to simplify that model and use references so that the double-lookup is avoided - I think ref makes sense in this situation, unless we make all API string-based (ie `send(id: peerid)` which is a bit nutty in nim